src/modules/Preferences/OrgPreferences/CultureAccordions/useCultureAccordions.tsx

```diff
@@ -0,0 +1,62 @@
+import { ChangeEvent, useState } from 'react';
+import { useLoaderData } from 'react-router-dom';
+import { PREFERENCES_CULTURE_QUESTIONS } from 'src/constants/PREFERENCES';
+import { Preference, PreferenceValue, updatePreferences } from 'src/core/api';
+import { removedEmptyProps } from 'src/core/utils';
+
+export const useCultureAccordions = () => {
+  const limitDescription = 160;
+  const { preferences } = useLoaderData() as { preferences: Preference[] };
+  const initialCultures = PREFERENCES_CULTURE_QUESTIONS.map(culture => {
+    const hasCulturePreferences = preferences.find(preference => preference.title === culture.key);
+    return hasCulturePreferences
+      ? { ...culture, value: hasCulturePreferences.value, description: hasCulturePreferences.description }
+      : culture;
+  });
+  const [currentCulturePreferences, setCurrentCulturePreferences] = useState(initialCultures);
+  const [letterCounts, setLetterCounts] = useState<Record<string, number> | null>(null);
+  const [errors, setErrors] = useState<Record<string, string>>({});
+
+  const onSelectCultureValues = (name: string, value: string) => {
+    const updatedCulturePreferences = currentCulturePreferences.map(culture =>
+      culture.key === name ? { ...culture, value: value as PreferenceValue } : culture,
+    );
+    setCurrentCulturePreferences(updatedCulturePreferences);
+  };
+
+  const handleCultureDescriptions = (name: string, value: string) => {
+    const updatedCulturePreferences = currentCulturePreferences.map(culture =>
+      culture.key === name ? { ...culture, description: value } : culture,
+    );
+    setCurrentCulturePreferences(updatedCulturePreferences);
+    setLetterCounts({ ...letterCounts, [name]: value.length });
+    if (value.length > limitDescription) setErrors({ ...errors, [name]: 'Too Long' });
+    else setErrors(removedEmptyProps({ ...errors, [name]: '' }) as Record<string, string>);
+  };
+
+  const onSubmit = async (e: ChangeEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    const preferences: Preference[] = currentCulturePreferences
+      .map(culture => {
+        //FIXME: API issue
+        const payload = removedEmptyProps({
+          title: culture.key,
+          value: culture.value as PreferenceValue,
+          description: culture.description,
+        });
+        return payload as Preference;
+      })
+      .filter(culture => culture.value);
+
+    try {
+      await updatePreferences({ preferences });
+    } catch (e) {
+      console.log('error in updating culture preferences', e);
+    }
+  };
```

Comment:
I think somewhere in mounting component, you should get the preferences again, since after updating, changing the tabs causes the preferences initialize with loader data again and it seems nothing has changed until refreshing the page